src/csp_route.c

```diff
@@ -223,7 +222,7 @@ int csp_route_work(uint32_t timeout) {
 		/* New incoming connection accepted */
 		csp_id_t idout;
 		idout.pri   = packet->id.pri;
-		idout.src   = my_address;
+		idout.src   = packet->id.dst;
```

Comment:
Please check that this behaves well when `packet->id.dst == CSP_BROADCAST_ADDR`.